consensus/phases/data.go

```diff
@@ -0,0 +1,224 @@
+/*
+ *    Copyright 2018 Insolar
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package phases
+
+import "github.com/damnever/bitarray"
+
+// ----------------------------------PHASE 1--------------------------------
+
+// PulseDataExt is a pulse data extension.
+type PulseDataExt struct {
+	NextPulseDelta uint16
+	PrevPulseDelta uint16
+	EpochPulseNo   uint32
+	PulseTimestamp uint32
+	OriginID       *bitarray.BitArray
+	Entropy        *bitarray.BitArray
+}
+
+// PulseData is a pulse data.
+type PulseData struct {
+	PulseNumer uint32
+	Data       *PulseDataExt
+}
+
+type NodePulseProof struct {
+	NodeStateHash *bitarray.BitArray
+	NodeSignature *bitarray.BitArray
+}
+
+func NewNodePulseProof() *NodePulseProof {
+	return &NodePulseProof{
+		NodeSignature: bitarray.New(512),
+		NodeStateHash: bitarray.New(512),
+	}
+}
+
+// NewPulseData creates and returns a pulse data.
+func NewPulseData() *PulseData {
+	return &PulseData{
+		Data: NewPulseDataExt(),
+	}
+}
+
+// NewPulseDataExt creates and returns a pulse data extension.
+func NewPulseDataExt() *PulseDataExt {
+	return &PulseDataExt{
+		OriginID: bitarray.New(128),
+		Entropy:  bitarray.New(256),
+	}
+}
+
+// --------------REFERENDUM--------------
+
+type ReferendumClaim interface {
+	Type() *bitarray.BitArray
+	Length() *bitarray.BitArray
+}
+
+// NodeBroadcast is a broadcast of info. Must be brief and only one entry per node.
+// Type 4.
```

Comment:
Should we store this TYPES in constants?

Reply:
yep. fixed